airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py

```diff
@@ -86,6 +89,43 @@ def transform(self, integration_type: DestinationType, config: Dict[str, Any]):
 
         return base_profile
 
+    @staticmethod
+    def is_ssh_tunnelling(config: Dict[str, Any]) -> bool:
+        tunnel_methods = ["SSH_KEY_AUTH", "SSH_PASSWORD_AUTH"]
+        if (
+            "tunnel_method" in config.keys() and 
+            "tunnel_method" in config["tunnel_method"] and 
+            config["tunnel_method"]["tunnel_method"].upper() in tunnel_methods
+        ):
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def is_port_free(port: int) -> bool:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            try:
+                s.bind(("localhost", port))
+            except Exception as e:
+                print(f"port {port} unsuitable: {e}")
+                return False
+            else:
+                print(f"port {port} is free")
+                return True
+
+    @staticmethod
+    def pick_a_port() -> int:
+        """
+        This function finds a free port, starting with 50001 and adding 1 until we find an open port.
+        """
+        port_to_check = 50001  # just past start of dynamic port range (49152:65535)
```

Comment:
going to skip doing this, for following reasons:
- would require converting (pretty much all of) the class methods to non-static so we can share a single attribute reference for chosen ssh port, to avoid getting a different value in dbt config v.s. written to file for script.
- the above work would be fine to do... except once [this issue](https://github.com/airbytehq/airbyte/issues/5091) is resolved, it would all be redundant anyway.